tests/unit/semver-intersect.js

```diff
@@ -138,6 +138,30 @@ describe('intersect', () => {
         const result = intersect('^4.0.0', '~4.3.0');
         expect(result).to.equal('~4.3.0');
     });
+    it('should handle pre-release versions mixed with versions', () => {
+        const result = intersect('^1.0.0-alpha.3', '^1.2.0');
+        expect(result).to.equal('^1.2.0');
+    });
+    it('should handle compatible pre-release versions', () => {
+        const result = intersect('^1.0.0-alpha.3', '^1.0.0-alpha.4');
+        expect(result).to.equal('^1.0.0-alpha.4');
+    });
+    it('should handle compatible pre-release versions without a dot', () => {
+        const result = intersect('^0.14.0-beta2', '^0.14.0-beta3');
+        expect(result).to.equal('^0.14.0-beta3');
+    });
+    it('should handle compatible pre-release versions with the same preid', () => {
+        const result = intersect('^0.14.0-beta', '^0.14.0-beta4');
+        expect(result).to.equal('^0.14.0-beta4');
+    });
+    it('should handle incompatible pre-release versions (different patch version)', () => {
+        const call = intersect.bind(null, '^1.9.0-alpha', '^1.9.1-alpha');
+        expect(call).to.throw('Prerelease version >=1.9.1-alpha is not compatible with >=1.9.0-alpha');
+    });
+    it('should handle incompatible pre-release versions (different preid)', () => {
+        const call = intersect.bind(null, '^1.9.0-alpha', '^1.9.0-beta');
+        expect(call).to.throw('Prerelease version >=1.9.0-beta is not compatible with >=1.9.0-alpha');
```

Comment:
That is not correct. `1.9.0-beta` **does** satisfy `^1.9.0-alpha` for the same reason that i mentioned above:

<img width="393" alt="bildschirmfoto 2018-02-01 um 14 27 18" src="https://user-images.githubusercontent.com/20770029/35680944-1b50f47a-075c-11e8-8916-7707824d29ce.png">